peer.go

```diff
@@ -517,6 +522,10 @@ func (p *Peer) connectionCloseStateChange(changed *Connection) {
 
 	if found {
 		p.onClosedConnRemoved(p)
+		// Inform third parties that a peer lost a connection.
+		if p.channel.onPeerStatusChanged != nil {
```

Comment:
Optionally: once we do a DI approach, we may want to set a dummy noop function if this is `nil` so we can avoid the nil checks.